mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -248,18 +248,18 @@ class ALSModel private[ml] (
   @Since("1.3.0")
   def setPredictionCol(value: String): this.type = set(predictionCol, value)
 
+  private val predict = udf { (userFeatures: Seq[Float], itemFeatures: Seq[Float]) =>
+    if (userFeatures != null && itemFeatures != null) {
+      blas.sdot(rank, userFeatures.toArray, 1, itemFeatures.toArray, 1)
```

Comment:
I wonder how the overhead of converting to an array compares with the efficiency of calling sdot -- could be faster to just do the Seqs by hand? is it possible to operate on something besides Seq?